packages/client/src/runtime/core/types/Utils.ts

```diff
@@ -118,6 +118,19 @@ export type PayloadToResult<P, O extends Record<any, any> = RenameAndNestPayload
 
 export type Select<T, U> = T extends U ? T : never
 
+export type Equals<A, B> = (<T>() => T extends A ? 1 : 2) extends <T>() => T extends B ? 1 : 2 ? 1 : 0
```

Comment:
Does it come from [here](https://github.com/microsoft/TypeScript/issues/27024#issuecomment-421529650)?

Reply:
I had it in ts-toobelt, but it really originally was created by a former Microsoft employee that worked on TS